projects/packages/scheduled-updates/tests/php/class-scheduled-updates-test.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * Test class for Scheduled_Updates.
+ *
+ * @package automattic/scheduled-updates
+ */
+
+/**
+ * Test class for Scheduled_Updates.
+ *
+ * @coversDefaultClass Scheduled_Updates
+ */
+class Scheduled_Updates_Test extends \WorDBless\BaseTestCase {
+
+	/**
+	 * Set up.
+	 *
+	 * @before
+	 */
+	protected function set_up() {
+		parent::set_up_wordbless();
+		\WorDBless\Users::init()->clear_all_users();
+
+		// Initialize the WordPress filesystem variable.
+		global $wp_filesystem;
+		require_once ABSPATH . 'wp-admin/includes/file.php';
+		WP_Filesystem();
+		$this->wp_filesystem = $wp_filesystem;
+
+		$this->plugin_dir = WP_PLUGIN_DIR;
+		$this->admin_id   = wp_insert_user(
+			array(
+				'user_login' => 'dumasdasdasmy_user',
+				'user_pass'  => 'dummy_pass',
+				'role'       => 'administrator',
+			)
+		);
+		wp_set_current_user( $this->admin_id );
+
+		// Ensure plugin directory exists.
+		$this->wp_filesystem->mkdir( $this->plugin_dir );
+
+		// init the hook
+		add_action( 'rest_api_init', array( 'Automattic\Jetpack\Scheduled_Updates', 'add_is_managed_extension_field' ) );
+
+		do_action( 'rest_api_init' );
+	}
+
+	/**
+	 * Clean up after test
+	 *
+	 * @after
+	 */
+	protected function tear_down() {
+		// Clean up the temporary plugin directory
+		$this->wp_filesystem->rmdir( $this->plugin_dir, true );
+
+		parent::tear_down_wordbless();
+	}
+
+	/**
+	 * Simulate and test both managed and unmanaged plugins
+	 *
+	 * @covers ::add_is_managed_extension_field
+	 */
+	public function test_mixed_plugins() {
```

Comment:
Not sure if it's better to keep it in one function or to split it up into two functions, maybe something like `test_unmanaged_plugins` and `test_managed_plugins` so it's more clear? What do you think?

Reply:
Updated in https://github.com/Automattic/jetpack/pull/36098/commits/a509b8731ac2de77f2b9ab7f0c69ab78e7ba4ae5

Thank you!